[contribution] Add release maintainer protocol section.

130s opened the pull request:
Making a release is getting more complexed and it's becoming so easy to make simple but significant mistakes. So to remind myself (and to hopefully make the whole process more visual for everyone) I describe the process.

130s commented:
Documenting the process like this (and sticking to it) will hopefully avoid the situation like https://github.com/ros-planning/moveit/issues/393#issuecomment-270611411 (where there was no soak time at all).

v4hn commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
+
+* We use the minor version to differentiate the releases from different development branches.
+  * As of December 2016, Indigo 0.7.x (indigo-devel branch), Jade 0.8.x (jade-devel), Kinetic 0.9.x (kinetic-devel).
+* For API/ABI disrupting changes, we often introduce them even into the distributions already released (unlike what's suggested [Release Policy on ROS wiki](http://wiki.ros.org/Distributions/ReleasePolicy#Planning_your_Development)) depending on the discussion among developers. When that happens, we carefully raise attention of the users but most likely won't add special version handling.
```
API *disrupting* changes should not happen. We worked quite a bit to make API changes compatible last year,
even if we mess up sometimes. This bullet point reads like we don't care.
ABI disrupting changes are handled via soname bumps. For kinetic this happens automatically on every release, but for indigo we try to avoid breakage at all and bump sonames explicitly if we really don't want to avoid the change.
At least that's the idea...

I would leave this bullet point out of this list..

davetcoleman replied:
> I would leave this bullet point out of this list..

I link this bullet, just remove the "API" part

130s replied:
Removed the entire line about api/abi breakage (sorry for the misunderstanding, I'm still figuring out about how we handle this).

davetcoleman commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
 If possible, merge these *together* with the original request.
 In this context it might be worth spending some time on making use of features available in later ROS distributions to simplify the code, e.g. by using a new coding standard or a more current version of a library.
+
+### Making a new release
```
Seems like this warrants a new top-level section:

> ## Releasing MoveIt! Into ROS Buildfarm with Bloom

davetcoleman replied:
(two hashtags instead of three)

davetcoleman commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
+#### Typical protocol for a single release
+If each step ends with issues, they need to be fixed before moving on.
+
+1. Notify maintainers to freeze new merge until you finish.
```
+1

130s replied:
Couldn't figure out how to systematically add "zero" enumeration item in the markup syntax. So I merged [the suggestion from @v4hn](https://github.com/ros-planning/moveit.ros.org/pull/116#discussion_r94870024) into the first bullet.

davetcoleman commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
+If each step ends with issues, they need to be fixed before moving on.
+
+1. Notify maintainers to freeze new merge until you finish.
+1. Plan the possible date where you finish releasing tasks. Then communicate with ROS buildfarm mainteners at https://discourse.ros.org/c/release to tell that we want soak time of one or two weeks so that maintainers/early adopters can test the binaries from [shadow repo](http://wiki.ros.org/ShadowRepository) and add necessary fix if any.
```
make Discourse link a hyperlink?

davetcoleman commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
+
+1. Notify maintainers to freeze new merge until you finish.
+1. Plan the possible date where you finish releasing tasks. Then communicate with ROS buildfarm mainteners at https://discourse.ros.org/c/release to tell that we want soak time of one or two weeks so that maintainers/early adopters can test the binaries from [shadow repo](http://wiki.ros.org/ShadowRepository) and add necessary fix if any.
+1. Run ROS buildfarm prerelease test for **all** supported Ubuntu distribution in [REP-0003](http://www.ros.org/reps/rep-0003.html) 
```
distribution should be plural

130s replied:
I added links to ROS wiki, where I just added an instruction for how to run the prerelease test (with some concrete examples using MoveIt! :P ).

davetcoleman commented on documentation/contributing/pullrequests/index.markdown:
```diff
@@ -86,3 +86,27 @@ If the commits apply cleanly to future branches and there is no evidence that th
+1. Run ROS buildfarm prerelease test for **all** supported Ubuntu distribution in [REP-0003](http://www.ros.org/reps/rep-0003.html) 
+  * As long as REP-0003 supports, we must test even EOLed Ubuntu distro (e.g. Saucy for ROS Indigo was retired in 2014 but REP-0003 still supports it and there's no way as of December 2016 to skip it. See [moveit/#100](https://github.com/ros-planning/moveit/issues/100)).
+1. Update changelogs. Take advantage of `catkin_generate_changelog` command to populate new logs, then preferably edit them manually to sort out per the type of changes (e.g. bugfix, new capability, maintenance, documentation).
+1. Create a new tag with an appropriate version number (see the version policy section). Utilize `catkin_prepare_release` command that bumps the versions in package xml and in changelog files, creates a new tag, and pushes it to the remote repo.
```
For the packages I release from my personal repos, I have saved the full command needed for these for faster release. Perhaps add those here too, at least for the ``moveit`` repo?

130s commented:
Thanks for the review, all comments should be now addressed and Travis passed. I've added many lines so taking yet another look would be the most appreciated.
Merger can squash commits.
